tests/suite/test_prometheus_exporter.py

```diff
@@ -19,40 +49,124 @@ def enable_exporter_port(cli_arguments, kube_apis,
-                                           'nginx_ingress_controller_nginx_reloads_total{class="nginx",reason="other"}'])
-                          ],
-                         indirect=["ingress_controller"])
 class TestPrometheusExporter:
-    def test_metrics(self, ingress_controller_endpoint, ingress_controller, enable_exporter_port, expected_metrics):
+    @pytest.mark.parametrize(
+        "ingress_controller, expected_metrics",
+        [
+            pytest.param(
+                {"extra_args": ["-enable-prometheus-metrics"]},
+                [
+                    'nginx_ingress_controller_nginx_reload_errors_total{class="nginx"} 0',
+                    'nginx_ingress_controller_ingress_resources_total{class="nginx",type="master"} 0',
+                    'nginx_ingress_controller_ingress_resources_total{class="nginx",type="minion"} 0',
+                    'nginx_ingress_controller_ingress_resources_total{class="nginx",type="regular"} 0',
+                    "nginx_ingress_controller_nginx_last_reload_milliseconds",
+                    'nginx_ingress_controller_nginx_last_reload_status{class="nginx"} 1',
+                    'nginx_ingress_controller_nginx_reload_errors_total{class="nginx"} 0',
+                    'nginx_ingress_controller_nginx_reloads_total{class="nginx",reason="endpoints"}',
+                    'nginx_ingress_controller_nginx_reloads_total{class="nginx",reason="other"}',
+                ],
+            )
+        ],
+        indirect=["ingress_controller"],
+    )
+    def test_metrics(
+        self,
+        ingress_controller_endpoint,
+        ingress_controller,
+        enable_exporter_port,
+        expected_metrics,
+    ):
+        req_url = f"http://{ingress_controller_endpoint.public_ip}:{ingress_controller_endpoint.metrics_port}/metrics"
+        ensure_connection(req_url, 200)
+        resp = requests.get(req_url)
+        assert resp.status_code == 200, f"Expected 200 code for /metrics but got {resp.status_code}"
+        resp_content = resp.content.decode("utf-8")
+        for item in expected_metrics:
+            assert item in resp_content
+
+    @pytest.mark.parametrize(
+        "ingress_controller, expected_metrics",
+        [
+            pytest.param(
+                {"extra_args": ["-enable-prometheus-metrics", "-enable-latency-metrics"]},
+                [
+                    'nginx_ingress_controller_workqueue_depth{class="nginx",name="taskQueue"}',
+                    'nginx_ingress_controller_workqueue_queue_duration_seconds_bucket{class="nginx",name="taskQueue",le=',
+                    'nginx_ingress_controller_workqueue_queue_duration_seconds_sum{class="nginx",name="taskQueue"}',
+                    'nginx_ingress_controller_workqueue_queue_duration_seconds_count{class="nginx",name="taskQueue"}',
+                    'nginx_ingress_controller_upstream_server_response_latency_ms_bucket{class="nginx",code="200",pod_name=',
```

Comment:
this might be a wishlist :) , but should we also check that  the labels are not empty or set to the expected values? class, code, pod_name, resource_name, resource_namespace,resource_type,server,service,upstream